xrmocap/data/data_converter/humman_smc_data_converter.py

```diff
@@ -0,0 +1,311 @@
+            self.bbox_detector = bbox_detector
+
+        if isinstance(kps2d_estimator, dict):
+            kps2d_estimator['logger'] = logger
+            self.kps2d_estimator = build_detector(kps2d_estimator)
+        else:
+            self.kps2d_estimator = kps2d_estimator
+        self.visualize = visualize
+        if self.visualize:
+            vis_percep2d = self.bbox_detector is not None and \
+                    self.kps2d_estimator is not None
+            self.visualization = MviewMpersonDataVisualization(
+                data_root=data_root,
+                meta_path=meta_path,
+                vis_percep2d=vis_percep2d,
+                output_dir=os.path.join(meta_path, 'visualize'),
+                verbose=verbose,
+                logger=logger)
+
+    def run(self, overwrite: bool = False) -> None:
+        """Convert xrmocap meta data from source dataset. Scenes will be
+        created, and there are image lists, XRPrimer cameras, ground truth
+        keypoints3d, perception2d in each of the scene. If any of
+        bbox_detector, kps2d_estimator is None, skip converting perception2d.
+        If visualize is checked, perception2d and ground truth will be
+        visualized.
+
+        Args:
+            overwrite (bool, optional):
+                Whether replace the files at
+                self.meta_path.
+                Defaults to False.
+        """
+        BaseDataCovnerter.run(self, overwrite=overwrite)
+        with open(os.path.join(self.meta_path, 'dataset_name.txt'),
+                  'w') as f_write:
+            f_write.write(f'{self.dataset_name}')
+        for scene_idx, smc_path in enumerate(self.smc_paths):
+            smc_reader = SMCReader(file_path=smc_path)
+            scene_dir = os.path.join(self.meta_path, f'scene_{scene_idx}')
+            os.makedirs(scene_dir, exist_ok=True)
+            self.covnert_smc_frames(scene_idx, smc_reader)
+            self.covnert_image_list(scene_idx, smc_reader)
+            self.convert_cameras(scene_idx, smc_reader)
+            if self.bbox_detector is not None and \
+                    self.kps2d_estimator is not None:
+                self.convert_perception_2d(scene_idx, smc_reader)
+        if self.visualize:
+            self.visualization.run(overwrite=overwrite)
+
+    def covnert_smc_frames(self, scene_idx: int,
+                           smc_reader: SMCReader) -> None:
+        """Convert smc in source dataset to extracted jpg pictures.
+
+        Args:
+            scene_idx (int):
+                Index of this scene.
+            smc_reader (SMCReader):
+                SMCReader of this scene.
+        """
+        scene_dir = os.path.join(self.meta_path, f'scene_{scene_idx}')
+        smc_path = self.smc_paths[scene_idx]
+        smc_name = os.path.basename(smc_path)
+        if self.verbose:
+            self.logger.info('Extracting image relative path list' +
+                             f' for scene {scene_idx}, file {smc_name}.')
```

Comment:
Is it possible for users to pick the camera, not in order?

Reply:
Fixed in [0ef32be](https://github.com/openxrlab/xrmocap/pull/123/commits/0ef32beaf28079eadfa7ebc968c8aef27dad3e3a) .